src/client/menu/menu.c

```diff
@@ -2915,9 +2915,37 @@ Mods_MenuInit(void)
     int currentmod;
     int x = 0;
     int y = 0;
+    char modname[MAX_QPATH]; //TG626
 
     Mods_NamesInit();
+	
+    // create array of bracketed display names from folder names - TG626
+    char **displaynames=malloc(sizeof(*displaynames) * nummods);
 
+    for (int i=0; i < nummods; i++)
+    {
+        strcpy(modname, "[");
+	if (strlen(modnames[i]) < 16)
```

Comment:
try something like
```c
if(strlen(modnames[i]) < 16) {
    snprintf(modname, sizeof(modname), "[%-16s]", modnames[i]);
} else {
    snprintf(modname, sizeof(modname), "[%.12s...]", modnames[i]);
}
```
(I didn't really test this, maybe 16 and 12 aren't the right numbers for the right width, but something like this should work)

Reply:
later, all use strcat, and therefore are null terminated.

Reply:
these two snprintf() calls (and one strlen()) are all you need, no strcat(), no strcpy()

Reply:
What I have is a working system, I see no clear advantage to _changing_ it to your method, more elegant tho it may be. 
If and when this gets committed to the master branch, feel free to implement your version - I won't be offended.

Reply:
hmm ok, I misremembered what was wrong with strncat() (it *does* null-terminate, but use the number of chars in src as argument instead of the buffer size, which is error-prone but not a problem in this case. I confused ith with **strncpy()** which does *not* guarantee null-termination if src has more than n chars), so yes, it should work, though it's a lot of code for what could be a one-liner :)